lua/clipboard-image/utils.lua

```diff
@@ -63,21 +63,41 @@ M.is_clipboard_img = function (content)
   return false
 end
 
----@param dir string
+
+---Check if resolve any complicated pathings
+---@param dir string or table
+M.resolve_dir = function(dir)
+  local this_os = M.get_os()
+  if (type(dir) == "table") then
+    local full_path = ""
+    for _, value in pairs(dir) do
+        if this_os == 'Windows' then
+          full_path = full_path .. "\\" .. vim.fn.expand(value)
+        else
+          full_path = full_path .. "/" .. vim.fn.expand(value)
+        end
+    end
+    return full_path
```

Comment:
Instead of using `if` to decide path separator, I think it's better if we add one more parameter to store its value. So it'll be like this

```suggestion
---@param dir string|table
---@param path_separator string
M.resolve_dir = function(dir, path_separator)
  if (type(dir) == "table") then
    path_separator = path_separator or '/'
    local full_path = ""
    for _, value in pairs(dir, path_separator) do
      full_path = full_path .. path_separator .. vim.fn.expand(value)
    end
    return full_path
```

Also I believe this method is better because we don't always need `\\` on Windows. We just use it for copying image but for inserting text we still use unix path separator